packages/app/components/onboarding/hcaptcha/hcaptcha-utils.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from "react";
+
+import { useRouter } from "@showtime-xyz/universal.router";
+
+import { useRedirectToClaimDrop } from "app/hooks/use-redirect-to-claim-drop";
+import { axios } from "app/lib/axios";
+
+export const useValidateCaptchaWithServer = () => {
+  async function validate(token?: string) {
+    if (!token) return Promise.reject(new Error("No token provided"));
+
+    return await axios({
+      url: "/v1/profile/captcha/verify",
+      method: "POST",
+      data: {
+        hcaptcha_response_token: token,
+      },
+    });
+  }
+
+  return { validate };
+};
+
+export const useFinishOnboarding = () => {
+  const router = useRouter();
+  const redirectToClaimDrop = useRedirectToClaimDrop();
+
+  const finishOnboarding = useCallback(
+    (redirectUri?: string) => {
+      console.log(redirectUri);
+      if (redirectUri) {
+        if (redirectUri.includes("/claim/")) {
```

Comment:
not a blocker: do you think we can handle it better if we implement that promise based approach?

await finishOnboarding() 😄

Reply:
I think we should definitely add that approach soon. We should tackle this in on rush and add the same for liking, clicking on comments etc. as we talked in our huddle. For now, leaving as is?